reactnative-todo/README.md

```diff
@@ -0,0 +1,242 @@
+
+IMPORTANT: This is not a demonstration of React Native itself or instructional for how to build a React Native app. This is intended to demonstrate the implementation of the ForgeRock JavaScript SDK within a React Native app. There are many aspects to routing, state management, tooling and other aspects to building a React Native app that are outside of the scope of this project. For information about creating a React Native app, visit React Native's official documentation.
+
+## Requirements
+
+1. An instance of ForgeRock's Access Manager (AM), either within a ForgeRock's Identity Cloud tenant, your own private installation or locally installed on your computer
+2. Node >= 14.2.0 (recommended: install via official package installer)
+3. Knowledge of using the Terminal/Command Line
+4. Ability to generate security certs
+5. This project "cloned" to your computer
+6. Xcode 12+
+
+## Installation
+
+Once you have the above requirements, we can build the project
+
+## Setup
+
+1. Fork this repo to your own personal account
+2. `git clone` from your personal fork
+3. `cd forgerock-react-native-sample`
+4. `cd reactnative-todo`
+5. `npm install`
+6. `cd ios`
+7. `pod install`
+8. `cd ..`
+9. Create the `.env.js` file from the `.env.example.js` file
+
+Once the above is done, start the build in Xcode using iPhone 12 or higher
+
+### Create Your OAuth Clients
+
+1. Create a public (native) OAuth client for the mobile app: no secret, scopes of openid profile email, implicit consent enabled, and no "token authentication endpoint method".
+2. Create a confidential (Node.js) OAuth client for the API server: with a secret, default scope of `am-introspect-all-tokens`, and `client_secret_basic` as the "token authentication endpoint method".
+
+### Create your Authentication Journeys/Trees
+
+1. Login
+2. Register
+
+Note: The sample app currently supports the following callbacks only:
+
+- NameCallback
+- PasswordCallback
+- ChoiceCallback
+- ValidatedCreateUsernameCallback
+- ValidatedCreatePasswordCallback
+- StringAttributeInputCallback
+- BooleanAttributeInputCallback
+- KbaCreateCallback
+- TermsAndConditionsCallback
+
+### Configure Your `.env.js` File
+
+To configure the React Native layer and the local Node.js API server, you'll need an environment file. To do this, change the name of `.env.js.template` to `.env.js` and add the relevant values for your setup.
+
+Example with annotations:
+
+```js
+const AM_URL = 'https://example-am-instance.forgerock.com/am'; // (include the /am)
+const API_PORT = 9443; // (your resource API server's PORT)
+const API_BASE_URL = 'https://localhost'; // (your resource API server's Base URL)
+const REALM_PATH = 'alpha'; // (AM realm; if using Identity Cloud, likely "alpha"; if not, likely "root")
+const REST_OAUTH_CLIENT = RestOAuthClient; // (name of private OAuth 2.0 client/application)
+const REST_OAUTH_SECRET = sf0h8e350efhd932; // (the secret for the private OAuth 2.0 client/application)
```

Comment:
This is a fake secret or something thats dead right?

Reply:
Yes